caffe2/serialize/inline_container.cc

```diff
@@ -42,27 +46,33 @@ static std::string basename(const std::string& name) {
 }
 
 size_t PyTorchStreamReader::read(uint64_t pos, char* buf, size_t n) {
-  in_->seekg(pos);
-  if(!*in_)
-    return 0;
-  in_->read(static_cast<char*>(buf), n);
-  if(!*in_)
-    return 0;
-  return n;
+  return in_->read(pos, buf, n, "reading file");
 }
 
-PyTorchStreamReader::PyTorchStreamReader(std::string file_name, std::istream* in)
-: ar_(new mz_zip_archive), in_(in) {
-  memset(ar_.get(), 0, sizeof(mz_zip_archive));
+PyTorchStreamReader::PyTorchStreamReader(const std::string& file_name)
+    : ar_(new mz_zip_archive),
+      in_(caffe2::make_unique<at::FileAdapter>(file_name)) {
+  init();
+}
 
-  if (!in_) {
-    file_stream_.open(file_name, std::ifstream::in | std::ifstream::binary);
-    in_ = &file_stream_;
-    valid("opening archive");
-  }
+PyTorchStreamReader::PyTorchStreamReader(std::istream* in)
+    : ar_(new mz_zip_archive),
```

Comment:
You can use `make_unique` for mz_zip_archive too.

Reply:
Yep, I should fix all the new.